contracts/core/src/interfaces/IFeeOracleV2.sol

```diff
@@ -9,81 +9,185 @@ import { IConversionRateOracle } from "./IConversionRateOracle.sol";
+    event DataGasPriceSet(uint64 dataCostId, uint64 gasPrice);
+
+    /// @notice Emitted when the base gas limit for a destination chain is set.
+    event BaseGasLimitSet(uint64 chainId, uint32 baseGasLimit);
 
-    /// @notice Emitted when the to-native conversion rate for a destination chain is set.
-    event ToNativeRateSet(uint64 chainId, uint64 toNativeRate);
+    /// @notice Emitted when the base data buffer for a data cost ID is set.
+    event BaseDataBufferSet(uint64 dataCostId, uint32 baseDataBuffer);
+
+    /// @notice Emitted when the data cost ID for a destination chain is set.
+    event DataCostIdSet(uint64 chainId, uint64 dataCostId);
+
+    /// @notice Emitted when the gas per byte for a data cost ID is set.
+    event GasPerByteSet(uint64 dataCostId, uint64 gasPerByte);
+
+    /// @notice Emitted when the to-native conversion rate for a gas token is set.
+    event ToNativeRateSet(uint8 gasToken, uint256 nativeRate);
+
+    /// @notice Emitted when the base protocol fee is set.
+    event ProtocolFeeSet(uint96 protocolFee);
 
     /// @notice Emitted when the manager is changed.
     event ManagerSet(address manager);
 
     /**
      * @notice Fee parameters for a specific chain.
+     * @custom:field gasToken       The gas token ID.
+     * @custom:field baseGasLimit   The base gas limit for that chain.
      * @custom:field chainId        The chain ID.
-     * @custom:field execGasPrice   The execution gas price on that chain (denominated in chains native token).
-     * @custom:field dataGasPrice   The data gas price on that chain (denominated in chains native token).
-     *                              ex. for Optimism, dataGasPrice is Ethereum L1's blob gas price.
-     * @custom:field toNativeRate   The conversion rate from the chains native token to this chain's
-     *                              native token. Rate is numerator over CONVERSION_RATE_DENOM.
+     * @custom:field gasPrice       The execution gas price on that chain (denominated in chains native token).
+     * @custom:field dataCostId     The data cost ID for that chain.
      */
     struct FeeParams {
+        uint8 gasToken;
+        uint32 baseGasLimit;
         uint64 chainId;
-        uint64 execGasPrice;
-        uint64 dataGasPrice;
-        uint64 toNativeRate;
+        uint64 gasPrice;
+        uint64 dataCostId;
+    }
+
+    /**
+     * @notice Data cost parameters for a data cost ID.
+     * @custom:field gasToken       The gas token ID.
+     * @custom:field baseDataBuffer The base data buffer for that data cost ID.
+     * @custom:field dataCostId     The data cost ID.
+     * @custom:field gasPrice       The data gas price for that data cost ID (denominated in chains native token).
+     * @custom:field gasPerByte     The gas per byte for that data cost ID.
+     */
+    struct DataCostParams {
+        uint8 gasToken;
+        uint32 baseDataBuffer;
+        uint64 dataCostId;
```

Comment:
Same thing here. I think we can drop `dataCostId` prefix because that namespaced by the struct typename.

Could just say `id`